### ISSUE TITLE
npm ERR! code ELIFECYCLE...

### ISSUE THREAD

**Issue by misterch0c** [balderdashy/sails#2824]
``` 
root@me:~# npm install -g sails
/
sails@0.11.0 preinstall /root/.nvm/v0.10.16/lib/node_modules/sails
node ./lib/preinstall_npmcheck.js

module.js:338
   throw err;
         ^
Error: Cannot find module '/root/.nvm/v0.10.16/lib/node_modules/sails/lib/preinstall_npmcheck.js'
   at Function.Module._resolveFilename (module.js:336:15)
   at Function.Module._load (module.js:278:25)
   at Function.Module.runMain (module.js:501:10)
   at startup (node.js:129:16)
   at node.js:814:3
npm ERR! Linux 3.8.0-29-generic
npm ERR! argv "node" "/root/.nvm/v0.10.16/bin/npm" "install" "-g" "sails"
npm ERR! node v0.10.16
npm ERR! npm  v2.7.5
npm ERR! code ELIFECYCLE 
```

This is driving me insane. It's a Ubuntu server and it's running other node apps already. So I can't install sails... Before I got this error though I succesfuly installed it but then when I ran 'sails' I got nothing. Tried to set my path, clean npm cache... Btw I was using npm 1.2 or so and sails require  < 1.4 so I'm running 2.7.4 now

**Comment by tjwebb** [comment]
See this issue: https://github.com/balderdashy/sails/issues/2673

tldr: don't run `npm install` as root.

**Comment by misterch0c** [comment]
Yes I saw that issue before posting, should have mentioned it. But when I don't run it as root I get this:
``` 
me@M4KE:~$ npm install sails -g
npm ERR! Linux 3.8.0-29-generic
npm ERR! argv "node" "/usr/local/bin/npm" "install" "sails" "-g"
npm ERR! node v0.12.2
npm ERR! npm  v2.7.5
npm ERR! path /usr/local/lib/node_modules/sails
npm ERR! code EACCES
npm ERR! errno -13

npm ERR! Error: EACCES, rmdir '/usr/local/lib/node_modules/sails'
npm ERR!     at Error (native)
npm ERR!  { [Error: EACCES, rmdir '/usr/local/lib/node_modules/sails']
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   path: '/usr/local/lib/node_modules/sails' }
npm ERR! 
npm ERR! Please try running this command again as root/Administrator.
npm ERR! error rolling back Error: EACCES, rmdir '/usr/local/lib/node_modules/sails'
npm ERR! error rolling back     at Error (native)
npm ERR! error rolling back  { [Error: EACCES, rmdir '/usr/local/lib/node_modules/sails']
npm ERR! error rolling back   errno: -13,
npm ERR! error rolling back   code: 'EACCES',
npm ERR! error rolling back   path: '/usr/local/lib/node_modules/sails' }

npm ERR! Please include the following file with any support request:
npm ERR!     /home/me/npm-debug.log
```

**Comment by tjwebb** [comment]
> npm ERR! Error: EACCES, rmdir '/usr/local/lib/node_modules/sails'

Yea, your permissions are jacked up from running npm as root.`chown` /usr/local/lib to your user

**Comment by misterch0c** [comment]
drwxr-xr-x 7 dav  root 4096 Apr  9 08:40 lib
I had already done that but the error stayed the same. It's not my server btw, but I'm supposed to get my app in prod on it. I guess it's some permission issues like you suggested. Before I created user "dav" there was only root on this server...

**Comment by diogoviannaaraujo** [comment]
Im trying to install sails on a plain 15.04 ubuntu with apt version of nodejs and npm (0.10.25 and 1.4.21) and im getting this errors, even if I don't run npm install -g sails as root.
